base/component_status.go

```diff
@@ -0,0 +1,205 @@
+package base
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/qri-io/dataset"
+	"github.com/qri-io/qfs/muxfs"
+	"github.com/qri-io/qri/base/component"
+	"github.com/qri-io/qri/base/dsfs"
+	"github.com/qri-io/qri/dsref"
+)
+
+var (
+	// STUnmodified is "no status"
+	STUnmodified = "unmodified"
+	// STAdd is an added component
+	STAdd = "add"
+	// STChange is a modified component
+	STChange = "modified"
+	// STRemoved is a removed component
+	STRemoved = "removed"
+	// STParseError is a component that didn't parse
+	STParseError = "parse error"
+	// STMissing is a component that is missing
+	STMissing = "missing"
+	// STConflictError is a component with a conflict
+	STConflictError = "conflict error"
+	// ErrWorkingDirectoryDirty is the error for when the working directory is not clean
+	ErrWorkingDirectoryDirty = fmt.Errorf("working directory is dirty")
+)
+
+// StatusItem is the status of a component of a dataset, and whether it
+// was changed at a specific version in history
+type StatusItem struct {
+	SourceFile string    `json:"sourceFile"`
+	Component  string    `json:"component"`
+	Type       string    `json:"type"`
+	Message    string    `json:"message"`
+	Mtime      time.Time `json:"mtime"`
```

Comment:
mtime is coming up empty for all fetches

Not sure if it's a bug, or just not needed any longer because we no longer rely on fsi?

Reply:
That is a good point. I believe `whatchanged` never returned mtime, because it would always match the commit's timestamp. It makes sense then to remove this field, I am also going to add a unit test.

Reply:
Also removed `SourceFile` and `Message`, as those are no longer needed without file system integration.